### PR TITLE
tidy up detestable detest

### PR DESCRIPTION
**Pull request by sydb**
Address issue #2267 in 2 separate commits as 2 separate steps in case folks like most of the ideas therein but do not want to see the detest.log file divided into 3 output log files.

### Note
I do not think this PR requires two separate reviewers.

If one of the 2 requested reviewers okays it, then one of the 3 assignees needs to EITHER a) review it with an eye towards the upcoming release — will this make upcoming release that much easier (because this part of the build is less confusing) or harder (because something is different), and merge if the former; OR b) deliberately punt it to next release as too difficult to review now.

### CODE REVIEW

File: P5/Test/Makefile [hunk 1]
```diff
@@ -107,19 +107,29 @@ detest: checkrnv
 	@echo " "
 	@echo "--------- work on target $@ ---------"
 	@echo BUILD: Test file with deliberate mistakes
+	@echo "--- run ant to validate ODD file (./detest.odd):"
+#	Note that output below is called 'detest_odd_schematron' because if there
+#	is a RelaxNG error the build dies and no output is generated
+	$(ANT) -f antruntest.xml -Doutputname=detest -DoddFile=detest.odd validateodd > detest_odd_schematron.log 2>&1
+	@echo "--- run ant to compile ODD file inoto RelaxNG and Schematron"
+	@echo "(detest.odd → detest.rng & detst.isosch):"
```
Comment (martindholmes) [inline P5/Test/Makefile]: Shouldn't that be "detest.isosch"?

Comment (sydb) [reply P5/Test/Makefile]: Yup.

Comment (sydb) [reply P5/Test/Makefile]: Still working on other bugs, will fix that typo, too, in a few mins.

**Comment by sydb** [comment]
OK. Took another crack at this. My apologies, but it turns out to be a pain to test, because when tested on my local system of course the paths match, so if you fail to delete them properly the result is the same!

**Comment by JanelleJenstad** [comment]
Bump @HelenaSabel or @martindholmes. Martin pointed out a typo in his review. We need one of you to okay the merge.

**Comment by sydb** [comment]
1. Typo @martindholmes spotted fixed in 48c7996.
2. It was just in an informative message, anyway. I.e., only those who build the GLs would ever have seen it.

**Comment by martindholmes** [comment]
Tests pass and in my checking they run faster than before. Nice work!

**Comment by sydb** [comment]
BTW, assignees, the thing you want to look at is the Makefile. No point in really paying attention to the log files, I don’t think. And you should not feel compelled (although you should feel free) to scrutinize every line of code. The main test is does it work when you use it to test the GLs.
The basic change is instead of validating in 3 different ways[1] generating one big log file, and then comparing it to expected, we now validate the same 3 different ways, but generate 3 separate log files (with semi-intelligent names) and compare each of them to expected individually. (This way, when there is a problem, you know which validation process it was in.) Also improved the messages you get as this happens so it is (a little) less confusing.

[1] 
1. validate detest.odd against p5odds.rng and p5odds.isosch (actually p5odds.message.isosch.xsl)
2. validate detest.xml against detest.rnc
3. validate detest.xml against detest.isosch (actually detest.xsl)

**Review by HelenaSabel — APPROVED** [review]
Thank you for the explanations, @sydb

I run the tests and indeed you notice the time difference!

**Pull request merged**